assets/js/modules/analytics/components/dashboard/DashboardPopularPagesWidget.js

```diff
@@ -0,0 +1,132 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from '../../datastore/constants';
+import { STORE_NAME as CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { STORE_NAME as CORE_MODULES } from '../../../../googlesitekit/modules/datastore/constants';
+import whenActive from '../../../../util/when-active';
+import PreviewTable from '../../../../components/preview-table';
+import Layout from '../../../../components/layout/layout';
+import { getDataTableFromData, TableOverflowContainer } from '../../../../components/data-table';
+import { numberFormat } from '../../../../util';
+import getDataErrorComponent from '../../../../components/notifications/data-error';
+import getNoDataComponent from '../../../../components/notifications/nodata';
+
+const { useSelect } = Data;
+
+function DashboardPopularPagesWidget() {
+	const {
+		analytics,
+		data,
+		error,
+		loading,
+	} = useSelect( ( select ) => {
+		const store = select( STORE_NAME );
+		const args = {
+			dateRange: select( CORE_USER ).getDateRange(),
+			dimensions: 'ga:pageTitle,ga:pagePath',
+			metrics: [
+				{
+					expression: 'ga:pageviews',
+					alias: 'Pageviews',
+				},
+			],
+			orderby: [
+				{
+					fieldName: 'ga:pageviews',
+					sortOrder: 'DESCENDING',
+				},
+			],
+			limit: 10,
+		};
+
+		return {
+			analytics: select( CORE_MODULES ).getModule( 'analytics' ),
+			data: store.getReport( args ),
+			error: store.getErrorForSelector( 'getReport', [ args ] ),
+			loading: store.isResolving( 'getReport', [ args ] ),
+		};
+	} );
+
+	if ( loading ) {
+		return <PreviewTable padding />;
+	}
+
+	if ( error ) {
+		return getDataErrorComponent( __( 'Analytics', 'google-site-kit' ), error.message );
+	}
+
+	if ( ! Array.isArray( data?.[ 0 ]?.data?.rows ) ) {
+		return getNoDataComponent( __( 'Analytics', 'google-site-kit' ) );
+	}
+
+	const headers = [
+		{
+			title: __( 'Most popular content', 'google-site-kit' ),
+			primary: true,
+		},
+		{
+			title: __( 'Views', 'google-site-kit' ),
+		},
+	];
+
+	const links = [];
+	const dataMapped = data[ 0 ].data.rows.map( ( row, i ) => {
+		const [ title, url ] = row.dimensions;
+		links[ i ] = url.startsWith( '/' ) ? url : '/' + url;
+
+		return [
+			title,
+			numberFormat( row.metrics[ 0 ].values[ 0 ] ),
+		];
+	} );
+
+	const options = {
+		hideHeader: false,
+		chartsEnabled: false,
+		links,
+		showURLs: true,
+	};
```

Comment:
We'll also need to add the change from #1922 to the new widget here, it should link to the single URL details view in the admin via `useAdminURLs`.

Reply:
Updated.